pkg/prometheus/resource_selector.go

```diff
@@ -1579,9 +1653,35 @@ func (rs *ResourceSelector) validateIonosSDConfigs(ctx context.Context, sc *moni
 			return fmt.Errorf("[%d]: %w", i, err)
 		}
 
+		if err := rs.validateCustomHTTPConfig(ctx, config.CustomHTTPConfig, rs.store, sc.GetNamespace()); err != nil {
+			return fmt.Errorf("[%d]: %w", i, err)
+		}
+
 		if err := rs.store.AddSafeTLSConfig(ctx, sc.GetNamespace(), config.TLSConfig); err != nil {
 			return fmt.Errorf("[%d]: %w", i, err)
 		}
 	}
 	return nil
 }
+
+func (rs *ResourceSelector) validateCustomHTTPConfig(ctx context.Context, hh monitoringv1.CustomHTTPConfig, store *assets.StoreBuilder, namespace string) error {
+	if reflect.ValueOf(hh).IsZero() || len(hh.HTTPHeaders) <= 0 {
+		return nil
+	}
+
+	if rs.version.LT(semver.MustParse("2.55.0")) {
+		return fmt.Errorf("HTTPHeaders is only supported for Prometheus version >= 2.55.0")
+	}
+
+	for _, v := range hh.HTTPHeaders {
```

Comment:
the operator also needs to ensure that no reserved header is referenced
https://github.com/prometheus/common/blob/0a89b987581831cf1f9061b3197b8d86f7c7b21c/config/headers.go#L27-L50

Reply:
Thanks @simonpasquier 
Got it
Add Validate done. Please re-check it